source/parser/for.civet

```diff
@@ -168,27 +168,55 @@ function processForInOf($0: [
         message: "'each' is only meaningful in for..of loops",
 
   // for own..in
-  let own = eachOwn and eachOwn[0].token is "own";
-  let expRef: ASTNode | undefined
+  own .= eachOwn and eachOwn[0].token is "own"
+  let expRef: ASTNode?
   if own and inOf.token is not "in"
     own = false
     eachOwnError =
       type: "Error"
       message: "'own' is only meaningful in for..in loops"
 
-  if !declaration2 and !own
+  // TypeScript doesn't support typed declarations in for loops,
+  // so pull such declarations inside the loop:
+  //   for var x: T of y -> for (const x1 of y) {var x: T = x1;
+  // for..in loops need a similar ref-based declaration
+  // for dereferencing to get the associated value.
+  { binding } := declaration
+  pattern .= binding?.pattern
+  if binding?.typeSuffix or (
+    inOf.token is "in" and declaration2 and pattern.type is not "Identifier"
+  )
+    keyRef := makeRef "key"
```

Comment:
Maybe we should just call it `"ref"` now since it may not actually be the key anymore.

Reply:
I renamed the Civet variable to `itemRef` (to still distinguish from `expRef`, for the thing we're iterating over) and renamed the ref itself to `"key"` or `"item"` depending on whether we're using `for..in` or `for..of` respectively.